Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryManager.java

```diff
@@ -154,39 +154,164 @@ else if (library != null) {
+            e.printStackTrace();
+        }
+
+
+        if (library != null &&
+                (librarySource.getLibraryContentType().equals(LibraryContentType.XML_ELM) ||
+                        translatorOptionsMatch(library, options))) {
+            translatedLibrary = generateTranslatedLibrary(library);
+            if (translatedLibrary != null) {
+                this.cacheLibrary(translatedLibrary);
+            }
+        }
+
+        return translatedLibrary;
+    }
+
+    private TranslatedLibrary generateTranslatedLibrary(Library library) {
+
+        if (library == null) {
+            return null;
+        }
+        boolean translationSuccess = true;
+        TranslatedLibrary translatedLibrary = new TranslatedLibrary();
+        try {
+            if (library != null) {
+                translatedLibrary.setLibrary(library);
+            }
+            if (library.getIdentifier() != null) {
+                translatedLibrary.setIdentifier(library.getIdentifier());
+            }
+
+            if (library.getUsings() != null && library.getUsings().getDef() != null) {
+                for (UsingDef usingDef : library.getUsings().getDef()) {
+                    translatedLibrary.add(usingDef);
+                }
+            }
+            if (library.getIncludes() != null && library.getIncludes().getDef() != null) {
+                for (IncludeDef includeDef : library.getIncludes().getDef()) {
+                    translatedLibrary.add(includeDef);
+                }
+            }
+            if (library.getCodeSystems() != null && library.getCodeSystems().getDef() != null) {
+                for (CodeSystemDef codeSystemDef : library.getCodeSystems().getDef()) {
+                    translatedLibrary.add(codeSystemDef);
+                }
+            }
+            for (ValueSetDef valueSetDef : library.getValueSets().getDef()) {
+                translatedLibrary.add(valueSetDef);
+            }
+
+            if (library.getCodes() != null && library.getCodes().getDef() != null) {
+                for (CodeDef codeDef : library.getCodes().getDef()) {
+                    translatedLibrary.add(codeDef);
+                }
+            }
+            if (library.getConcepts() != null && library.getConcepts().getDef() != null) {
+                for (ConceptDef conceptDef : library.getConcepts().getDef()) {
+                    translatedLibrary.add(conceptDef);
+                }
+            }
+            if (library.getParameters() != null && library.getParameters().getDef() != null) {
+                for (ParameterDef parameterDef : library.getParameters().getDef()) {
+                    translatedLibrary.add(parameterDef);
+                }
+            }
+            if (library.getStatements() != null && library.getStatements().getDef() != null) {
+                for (ExpressionDef expressionDef : library.getStatements().getDef()) {
+                    translatedLibrary.add(expressionDef);
+                }
```

Comment:
One of the issues with running from ELM is that the operator maps may not be populated correctly. There should be some code here to ensure that, and some tests to that effect as well.

Reply:
I believe those checking are inside TranlatedLibrary.
![image](https://user-images.githubusercontent.com/31050924/132012356-841726b6-3099-44c5-8c73-812246cd2fb6.png)
I can add some test. Is that ok?

Reply:
Yep.